nest/src/token/pUSD.sol

```diff
@@ -6,137 +6,493 @@ import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/I
+     * @return bool indicating whether the transfer was successful
+     */
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) public virtual override(ERC20Upgradeable, IERC20) nonReentrant returns (bool) {
+        address spender = _msgSender();
+        _spendAllowance(from, spender, amount);
+        _transfer(from, to, amount);
+        return true;
+    }
+
+    /**
+     * @notice Get the balance of shares for an account
+     * @param account The address to check the balance for
+     * @return The number of shares owned by the account
+     */
     function balanceOf(
         address account
-    ) public view override returns (uint256) {
-        return vault.balanceOf(account);
+    ) public view override(IERC20, ERC20Upgradeable) returns (uint256) {
+        pUSDStorage storage $ = _getpUSDStorage();
+        return $.boringVault.lens.balanceOf(account, $.boringVault.vault);
+    }
+
+    /**
+     * @notice Get the balance in terms of assets for an account
+     * @param account The address to check the balance for
+     * @return The value of shares in terms of assets owned by the account
+     */
+    function assetsOf(
+        address account
+    ) public view virtual override(ComponentToken) returns (uint256) {
+        pUSDStorage storage $ = _getpUSDStorage();
+        return $.boringVault.lens.balanceOfInAssets(account, $.boringVault.vault, $.boringVault.accountant);
+    }
+
+    // ========== METADATA OVERRIDES ==========
+
+    /**
+     * @notice Get the number of decimals for the token
+     * @return Number of decimals (6)
+     */
+    function decimals() public pure override(ERC4626Upgradeable, ERC20Upgradeable, IERC20Metadata) returns (uint8) {
+        return 6;
+    }
+
+    /**
+     * @notice Get the name of the token
+     * @return Name of the token
+     */
+    function name() public pure override(ERC20Upgradeable, IERC20Metadata) returns (string memory) {
+        return "Plume USD";
+    }
+
+    /**
+     * @notice Get the symbol of the token
+     * @return Symbol of the token
+     */
+    function symbol() public pure override(ERC20Upgradeable, IERC20Metadata) returns (string memory) {
+        return "pUSD";
     }
 
-    // ========== INTERFACE SUPPORT ==========
+    /**
+     * @notice Check if the contract supports a given interface
+     * @param interfaceId Interface identifier to check
+     * @return bool indicating whether the interface is supported
+     */
     function supportsInterface(
         bytes4 interfaceId
-    ) public view override(AccessControlUpgradeable) returns (bool) {
-        return super.supportsInterface(interfaceId);
+    ) public view virtual override(AccessControlUpgradeable, ComponentToken) returns (bool) {
+        return interfaceId == type(IERC20).interfaceId || interfaceId == type(IAccessControl).interfaceId
+            || super.supportsInterface(interfaceId);
```

Comment:
Could you move this into `ComponentToken.sol`? I think it already supports `IAccessControl` from inheritance but not `IERC20`